kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/tokenstore/TokenStoreState.java

```diff
@@ -0,0 +1,310 @@
+                logger.debug("received: {} records", records.count());
+                records.forEach(this::update);
+                if (!endOffsets.isEmpty()) {
+                    List<TopicPartition> toRemove = new ArrayList<>();
+                    endOffsets.forEach((tp, o) -> {
+                        long current = consumer.position(tp);
+                        if (current >= o) {
+                            toRemove.add(tp);
+                        }
+                    });
+                    toRemove.forEach(endOffsets::remove);
+                    if (endOffsets.isEmpty()) {
+                        isReady.get().complete(true);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            isReady.getAndUpdate(c -> {
+                c.complete(false);
+                return new CompletableFuture<>();
+            });
+            logger.warn("Error consuming to update Kafka token store", e);
+            if (isRunning.get()) {
+                logger.info("Restarting consumer, the model is kept, so errors updating state are expected");
+                executor.execute(this::startConsumer);
+            }
+        }
+    }
+
+    private void createTopic() {
+        try (AdminClient client = AdminClient.create(consumerConfiguration)) {
+            short replicationFactor = (short) Math.min(3, client.describeCluster().nodes().get().size());
+            NewTopic newTopic = new NewTopic(topic, 1, replicationFactor);
+            Map<String, String> topicConfig = new HashMap<>();
+            topicConfig.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT);
+            //needed to make sure we have at least one new message in case of concurrent writes
+            topicConfig.put(
+                    TopicConfig.MIN_COMPACTION_LAG_MS_CONFIG,
+                    String.valueOf((2 * Duration.from(claimTimeout).getSeconds()))
+            );
+            topicConfig.put(
+                    TopicConfig.MAX_COMPACTION_LAG_MS_CONFIG,
+                    String.valueOf((8 * Duration.from(claimTimeout).getSeconds()))
+            );
+            newTopic.configs(topicConfig);
+            client.createTopics(Collections.singletonList(newTopic));
+        } catch (ExecutionException | InterruptedException e) {
+            logger.warn("error creating topic for Kafka token store", e);
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private void update(ConsumerRecord<String, TokenUpdate> consumerRecord) {
+        boolean result;
+        TokenUpdate update;
+        if (consumerRecord.value() != null) {
+            update = consumerRecord.value();
+            result = updateAddition(update);
+        } else {
+            update = new TokenUpdate(consumerRecord.headers(), null);
+            result = updateDeletion(update);
+        }
+        feedback.computeIfPresent(
```

Comment:
Just to confirm, but you're doing `computeIfPresent` because the update you receive might come from a `KafkaTokenStore` instance on another node, correct?